task7/main.js

```diff
@@ -0,0 +1,508 @@
+    createdAt: new Date('2020-03-19T14:00:00'),
+    author: 'Весна',
+    photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg',
+    hashTags: ['#world', '#happiness', '#love', '#BSU'],
+    likes: ['Лера Конфетка', 'Саша Привалов', 'Полина Филипович']
+    },
+
+    {
+
+    id: '18',
+    description: 'Хочу быть Петровой',
+    createdAt: new Date('2020-06-19T03:00:00'),
+    author: 'Иванова Мария',
+    photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg',
+    hashTags: ['#world',  '#love', '#BSU'],
+    likes: ['Лера Конфетка', 'Саша Привалов', 'Полина Филипович']
+    },
+
+    {
+
+    id: '19',
+    description: 'спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+    createdAt: new Date('2030-03-14T22:00:00'),
+    author: 'Полина Лелис',
+    photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg',
+    hashTags: ['#world', '#happiness', '#love', '#BSU'],
+    likes: ['Лера Конфетка', 'Саша Привалов', 'Полина Филипович']
+    },
+
+    {
+
+    id: '20',
+    description: 'Любите себя сейчас',
+    createdAt: new Date('2001-03-15T03:00:00'),
+    author: 'Полина Лелис',
+    photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg',
+    hashTags: ['#world',  '#love', '#BSU'],
+    likes: ['Лера Конфетка', 'Саша Привалов', 'Полина Филипович']
+    },
+
+  ];
+
+class PostList{
+
+  _postCollection = [];
+
+  constructor(postCollection) {
+    this._postCollection = postCollection;
+  }
+
+        static _validatePost(post){
+          return ((Array.isArray(post.likes))&&
+            (post.createdAt instanceof Date)&&
+            (typeof post.id === 'string')&&
+            (typeof post.description === 'string')&&
+            (typeof post.author === 'string'));
+        }
+
+        _getPostIndexById(idElement) {
+            return this._postCollection.findIndex((post) => post.id === idElement);
+        }
+
+
+        addPost(post){
+          if (typeof post != 'object') {
+              return false;
+            }
+          if(PostList._validatePost(post)){
+              this._postCollection.push(post);
+              return true;
+            }
+          else {
+              return false;
+            }
+        }
+
+
+
+        getPosts(skip, top, filterConfig) {
+          let newPosts = this._postCollection.slice();
+          if (typeof filterConfig.author === 'string') {
+            newPosts = newPosts.filter(item => item.author === filterConfig.author);
+          }
+          
+          if (filterConfig.createdAt instanceof Date) {
+            newPosts = newPosts.filter(item => ((item.createdAt > filterConfig.createdAt - 1260601000)&&(item.createdAt < filterConfig.createdAt + 1260601000)));
+          }
+          if (Array.isArray(filterConfig.hashTags)) {
+            let valid = true;
+            for (let i of filterConfig.hashTags) {
```

Comment:
Should we check it?
If yes we can move it to the separate method.

Reply:
Btw: that method will be method most likely private as it does not sounds helpful outside.